rs/ic_os/os_tools/hostos_tool/src/main.rs

```diff
@@ -56,78 +56,116 @@ pub fn main() -> Result<()> {
 
     match opts.command {
         Some(Commands::GenerateNetworkConfig { output_directory }) => {
-            let config_map = config_map_from_path(Path::new(&opts.config))
-                .context("Please specify a valid config file with '--config'")?;
+            let config_map = config_map_from_path(Path::new(&opts.config)).context(format!(
+                "Failed to get config.ini settings for path: {}",
+                &opts.config
+            ))?;
             eprintln!("Using config: {:?}", config_map);
 
             let network_info = NetworkInfo::from_config_map(&config_map)?;
             eprintln!("Network info config: {:?}", &network_info);
 
-            let deployment_settings = get_deployment_settings(Path::new(&opts.deployment_file));
+            let deployment_settings = get_deployment_settings(Path::new(&opts.deployment_file))
+                .context(format!(
+                    "Failed to get deployment settings for file: {}",
+                    &opts.deployment_file
+                ))?;
+            eprintln!("Deployment config: {:?}", deployment_settings);
 
-            let deployment_name: Option<&str> = match &deployment_settings {
-                Ok(deployment) => Some(deployment.deployment.name.as_str()),
-                Err(e) => {
-                    eprintln!("Error retrieving deployment file: {e}. Continuing without it");
-                    None
+            let mgmt_mac = match deployment_settings.deployment.mgmt_mac {
+                Some(config_mac) => {
+                    let mgmt_mac = FormattedMacAddress::try_from(config_mac.as_str())?;
+                    eprintln!(
+                        "Using mgmt_mac address found in deployment.json: {}",
+                        mgmt_mac.get()
```

Comment:
nit: `get` just grabs the internal `String`, maybe it would be trivial to add `Display`?

Reply:
Added Display for UnformattedMacAddress and FormattedMacAddress!